packages/sdk/src/Actions/EventAction.ts

```diff
@@ -482,6 +483,8 @@ export class EventAction extends DeployableTarget<
   /**
    * Derives the action claimant address from a transaction based on the provided ActionClaimant configuration.
    * This method supports both event-based and function-based claimant derivation.
+   * **Important**: The claimant is considered to be `msg.sender` or `transaction.from` when `claimant.signature` is `zeroHash` or `claimant.fieldIndex` is 255.
```

Comment:
Let's tighten this up to explicitly `transaction.from` since it's technically not `msg.sender`